ktor-server/ktor-server-core/jvm/src/io/ktor/application/ApplicationEvents.kt

```diff
@@ -4,67 +4,15 @@
 
 package io.ktor.application
 
-import io.ktor.util.*
-import io.ktor.util.internal.*
-import kotlinx.coroutines.*
-
 /**
  * Provides events for [Application] lifecycle
  */
-public class ApplicationEvents {
-    private val handlers = CopyOnWriteHashMap<EventDefinition<*>, LockFreeLinkedListHead>()
-
-    /**
-     * Subscribe [handler] to an event specified by [definition]
-     */
-    public fun <T> subscribe(definition: EventDefinition<T>, handler: EventHandler<T>): DisposableHandle {
-        val registration = HandlerRegistration(handler)
-        handlers.computeIfAbsent(definition) { LockFreeLinkedListHead() }.addLast(registration)
-        return registration
-    }
-
-    /**
-     * Unsubscribe [handler] from an event specified by [definition]
-     */
-    public fun <T> unsubscribe(definition: EventDefinition<T>, handler: EventHandler<T>) {
-        handlers[definition]?.forEach<HandlerRegistration> {
-            if (it.handler == handler) it.remove()
-        }
-    }
-
-    /**
-     * Rise an event specified by [definition] with the specified [value] and call all handlers
-     *
-     * Handlers are called in order of subscriptions.
-     * If some handler throws an exception, all handlers will still run and then exception will be rethrown.
-     */
-    public fun <T> raise(definition: EventDefinition<T>, value: T) {
-        var exception: Throwable? = null
-        handlers[definition]?.forEach<HandlerRegistration> { registration ->
-            try {
-                @Suppress("UNCHECKED_CAST")
-                (registration.handler as EventHandler<T>)(value)
-            } catch (e: Throwable) {
-                exception?.addSuppressed(e) ?: run { exception = e }
-            }
-        }
-        exception?.let { throw it }
-    }
-
-    private class HandlerRegistration(val handler: EventHandler<*>) : LockFreeLinkedListNode(), DisposableHandle {
-        override fun dispose() {
-            remove()
-        }
-    }
-}
+public typealias ApplicationEvents = io.ktor.events.Events
```

Comment:
Not sure, should these declarations be deprecated?

Reply:
Yep, let's make deprecations